### DISCUSSION TITLE
Apa yang dapat menyebabkan OAuth2Token Error 400?

### DISCUSSION THREAD

**Discussion by revan2159** [ivanwilliammd/satusehat-integration#131] [Q&A (Tanya Jawab)] [answered]
**Describe the bug**
Attempt to read property "issue" on null
} catch (ClientException $e) {
            $res = json_decode($e->getResponse()->getBody()->getContents());
            **$issue_information = $res->issue[0]->details->text;  // NULL**
            $this->log($issue_information, 'POST Token', $url, null, (array) $res);
            return false;
        }

**Comment by SyaefulKai** [comment]
Error tsb bisa terjadi karena nilai $res->issue[0] bernilai NULL, yang dimana seharusnya $res->issue[0] ini berisikan pesan error dari request.
![Screenshot 2024-05-04 164034](https://github.com/ivanwilliammd/satusehat-integration/assets/158563298/0827e4fb-93d8-4250-93cc-50a32b352bb8)

**Comment by revan2159** [comment]
tapi respon nya tu 403 

![image](https://github.com/ivanwilliammd/satusehat-integration/assets/64825576/980110f6-fb1d-490a-980e-6bd719620c9e)

**Comment by revan2159** [comment]
error nya hilang jika 

`$headers = [
            'User-Agent' => 'PostmanRuntime/7.37.3',
            'Content-Type' => 'application/x-www-form-urlencoded'
        ];`

**Comment by ivanwilliammd** [comment, marked as answer]
Saya tutup issue ini, karena temuan ini invalid. saran saya coba perhatikan apakah : 
1. Internet / server hosting yg digunakan merupakan provider Indonesia, karena biasanya kalau tdk muncul error, bisa saja kena geoblocking
2. Ada kemungkinan kelebihan extraspace pada client ID & client secret sehingga throwing error
